object-construction-checker/tests/mustcall/MustCallChoiceLayeredStreams.java

```diff
@@ -0,0 +1,19 @@
+// Test case based on an MCC situation in Zookeeper.
+import org.checkerframework.checker.objectconstruction.qual.*;
+import org.checkerframework.checker.mustcall.qual.*;
+import org.checkerframework.checker.calledmethods.qual.*;
+import java.io.*;
+
+class MustCallChoiceLayeredStreams {
+    InputStream cache;
+    public InputStream createInputStream(String filename) throws FileNotFoundException {
+        if (cache == null) {
+            // The real version of this uses a mix of JDK and custom streams, so it makes more sense...
+            // TODO we shouldn't report a warning here and the code is okay because the cache is non-owning,
+            // and the caller of createInputStream is the owner of all of these streams.
+            // :: error: required.method.not.called
+            cache = new DataInputStream(new BufferedInputStream(new FileInputStream(new File(filename))));
```

Comment:
The problem here is the dataflow through the non-owning field, right?

Reply:
We don't track non-owning fields. If it was non-final owning field, then the problem was this.